c10/core/DispatchKey.h

```diff
@@ -106,31 +137,15 @@ enum class DispatchKey : uint8_t {
   // the corresponding dense tensors, and must be handled before them.
   MkldnnCPU, // registered at build/aten/src/ATen/RegisterMkldnnCPU.cpp
   // NB: not to be confused with MKLDNN, which is Caffe2 only
-  SparseCPU, // registered at build/aten/src/ATen/RegisterSparseCPU.cpp
-  SparseCUDA, // registered at build/aten/src/ATen/RegisterSparseCUDA.cpp
-  SparseHIP, // TODO: I think this is not actually used, due to Note
-  // [Masquerading as CUDA]
-  SparseXPU, // For out of tree Intel's heterogeneous computing plug-in
-  SparseVE, // For out of tree & closed source integration of SX-Aurora / NEC
+
+  // See [Note: Per-Backend Functionality Dispatch Keys]
+  Sparse,
 
   SparseCsrCPU,
   SparseCsrCUDA,
 
   NestedTensor, // lives out of tree at https://github.com/pytorch/nestedtensor
 
-  // Here are reserved backends for user-defined backends, see Note [Private use
-  // DispatchKey]
-  // To see some example about how to use this, check out ORT
-  PrivateUse1,
-  PrivateUse2,
-  PrivateUse3,
-
-  // Define an alias key to represent end of backend dispatch keys.
-  // If you add new backend keys after PrivateUse3, please also update it here.
-  // (But you shouldn't: private use keys should have higher precedence than
-  // all built-in keys)
-  EndOfBackendKeys = PrivateUse3,
```

Comment:
The DispatchKey enum starts with the functionality keys now, since the backend bits live in a different enum. I also have a semantic marker for when we're finished with the functionality bits and get to the "per-backend" runtime keys: `EndOfFunctionalityKeys`.

So `0 -> EndOfFunctionalityKeys` correspond to the individual functionality bits. `EndOfFunctionalityKeys+1 -> EndOfAutogradBackends` correspond to all of the per-backend runtime functionality keys, and `EndOfAutogradBackends+1 -> EndOfAliasKeys` correspond to alias keys.

Reply:
From offline discussion, since we have the fake backends at the bottom of dispatch key, we do want such marker.